src/drivers/QtTestDriver.cpp

```diff
@@ -4,28 +4,81 @@
 #include <QTextStream>
 #include <QTemporaryFile>
 
+#include <iostream>
+
 namespace cucumber {
 namespace internal {
 
+// wraps the QTemporaryFile creation
+// on Windows the file could not be written as long as QTemporaryFile owner of the file.
+
+class TemporaryFileWrapper {
+public:
+    static TemporaryFileWrapper create() {
+        QTemporaryFile tempFile(QString("%1/%2_%3")
+                                    .arg(
+                                        QDir::tempPath(),
+                                        qApp->applicationName().isEmpty() ? "qt_temp"
+                                                                          : qApp->applicationName()
+                                    )
+                                    .arg(qApp->applicationPid()));
+
+        if (!tempFile.open()) {
+            return {};
+        }
+
+        return {tempFile.fileName() + ".txt"};
```

Comment:
We may have a misunderstand here. According to the QTemporaryFile documentation for Qt5 and Qt6, the default constructor (without arguments) is guaranteed to create a unique filename. I think we should use this because creating unique filenames can be difficult, and I guess Qt did it good. It is ok for me when the filename is different between runs and Qt versions, as long as the content is deterministic (what it is by now).

My suggestion is:
```C++
    static TemporaryFileWrapper create() {
        QTemporaryFile tempFile();

        if (!tempFile.open()) {
            return {};
        }

        tempFile.setAutoRemove(false); // this is important if we remove the .txt extension! Otherwise the temp file is deleted at the end of this function. Remove this comment before committing :)

        return tempFile.fileName();
    }
```